### PR TITLE
add yum & alpine based linux support

### PR DESCRIPTION
**Pull request by gdams**
fixes:

```
sudo: apt-get: command not found
Error: The process '/usr/bin/sudo' failed with exit code 1
```

### CODE REVIEW

File: src/runaqa.ts [hunk 1]
```diff
@@ -96,7 +102,11 @@ export async function runaqaTest(
 }
 
 function getTestJdkHome(version: string, jdksource: string): string {
-  let javaHome = process.env[`JAVA_HOME_${version}_X64`] as string
+  // Try JAVA_HOME first and then fall back to GITHUB actions default location
+  let javaHome = process.env['JAVA_HOME'] as string
+  if (javaHome === undefined) {
+    javaHome = process.env[`JAVA_HOME_${version}_X64`] as string
```
Comment (sophia-guo) [inline src/runaqa.ts:108]: javaHome is the JDK image home we are aiming to test, which depends on the jdksource parameter. It differs from the default JAVA_HOME. So we can't try JAVA_HOME first and fall back to other location.

Comment (gdams) [reply src/runaqa.ts:108]: hmm okay, I can switch the logic round

File: src/runaqa.ts [hunk 2]
```diff
@@ -96,7 +102,11 @@ export async function runaqaTest(
 }
 
 function getTestJdkHome(version: string, jdksource: string): string {
+  // Try JAVA_HOME first and then fall back to GITHUB actions default location
   let javaHome = process.env[`JAVA_HOME_${version}_X64`] as string
+  if (javaHome === undefined) {
+    javaHome = process.env['JAVA_HOME'] as string
```
Comment (sophia-guo) [inline src/runaqa.ts:108]: If javaHome is undefined and jdksource === 'github-hosted' the test should stop with error that target JDK is not available.

Comment (gdams) [reply src/runaqa.ts:108]: done

**Review by sophia-guo — APPROVED** [review]
LGTM

**Pull request merged**